docs/internals/optimizer.rst

```diff
@@ -114,7 +114,7 @@ the instruction doesn't write to a certain location.
 
 For example,
 
-::
+.. code-block:: solidity
```

Comment:
```suggestion
.. code-block:: yul
```

Most (though not all) snippets in this file contiain Yul code rather than Solidity.

Reply:
It's my lack of experience in Yul language, it's pretty much similar to the Assembly code in Solidity. So that's why I regarded them as Solidity.

I will check this script now.

Reply:
My point of view for choosing `.. code-block:: solidity` rather than using `.. code-block:: yul` is to have the code highlited in Github, since using `yul` won't highlight the code.

Below is the difference between using both: 
- `.. code-block:: solidity`
![pic-selected-210608-0736-26](https://user-images.githubusercontent.com/36774944/121129310-0d783f80-c82d-11eb-8b9d-1acdb081710f.png)

- `.. code-block:: yul`
![pic-selected-210608-0736-47](https://user-images.githubusercontent.com/36774944/121129335-1537e400-c82d-11eb-8d0d-958e6978605f.png)

Reply:
I understand but will also switch to this inferior highlighting in the official docs (https://docs.soliditylang.org) where we do have proper Yul highlighting already and I think that takes priority.

Reply:
Alrigth great I will switch it to `yul`